public/plugin.ts

```diff
@@ -28,70 +28,67 @@ import { AppState } from './react-services/app-state';
-        // Set the flag in the telemetry saved object as the notice was seen and dismissed
-        this.hideTelemetryBanner && await this.hideTelemetryBanner();
-        
-        setScopedHistory(params.history);
-        // Load application bundle
-        const { renderApp } = await import('./application');
-        // Get start services as specified in kibana.json
-        const [coreStart, depsStart] = await core.getStartServices();
-        setErrorOrchestrator(ErrorOrchestratorService);
-        setHttp(core.http);
-        setCookies(new Cookies());
-        if(!AppState.checkCookies() || params.history.parentHistory.action === 'PUSH') {
-          window.location.reload();
-        }
-
-        await this.initializeInnerAngular();
-
-        //Check is user has Wazuh disabled
-        const response = await core.http.get(`/api/check-wazuh`);
-
-        params.element.classList.add('dscAppWrapper', 'wz-app');
-        const unmount = await renderApp(innerAngularName, params.element);
-
-        //Update if user has Wazuh disabled
-        this.stateUpdater.next(() => {
-          if (response.isWazuhDisabled) {
-            unmount();
+        try {
+          if (!this.initializeInnerAngular) {
+            throw Error('Wazuh plugin method initializeInnerAngular is undefined');
           }
-
-          return {
-            status: response.isWazuhDisabled,
-            category: {
-              id: 'wazuh',
-              label: 'Wazuh',
-              order: 0,
-              euiIconType: core.http.basePath.prepend(response.logoSidebar ? getAssetURL(response.logoSidebar) : getThemeAssetURL('icon.svg', UI_THEME)),
-            }}
-        })
-        return () => {
-          unmount();
-        };
+          // hide the telemetry banner.
+          // Set the flag in the telemetry saved object as the notice was seen and dismissed
+          this.hideTelemetryBanner && await this.hideTelemetryBanner();
+          setScopedHistory(params.history);
+          // Load application bundle
+          const { renderApp } = await import('./application');
+          // Get start services as specified in kibana.json
+          const [coreStart, depsStart] = await core.getStartServices();
+          setErrorOrchestrator(ErrorOrchestratorService);
+          setHttp(core.http);
+          setCookies(new Cookies());
+          if(!AppState.checkCookies() || params.history.parentHistory.action === 'PUSH') {
+            window.location.reload();
+          }
+          await this.initializeInnerAngular();
+          //Check is user has Wazuh disabled
+          const response = await WzRequest.genericReq(
+            'GET',
+            `/api/check-wazuh`,
+          )
+          params.element.classList.add('dscAppWrapper', 'wz-app');
+          const unmount = await renderApp(innerAngularName, params.element);
+          //Update if user has Wazuh disabled
+          this.stateUpdater.next(() => {
+            if (response.isWazuhDisabled) {
```

Comment:
**issue**: the `response` variable doesn't have the endpoint response directly when using `WzRequest.genericReq` service, so it will not work as expected.

Reply:
This comment was resolved with this commit [8be9602](https://github.com/wazuh/wazuh-kibana-app/pull/4362/commits/8be960288a82421ccff10942a6e56de52ab5a956)